### PR TITLE
Fix gradient population for FV kernels through FunctorElementalAux

### LINKED ISSUE #23146: Need capability to populate AuxVariableFE<VectorValue> with  Gradient of MooseVariableFVReal
## Reason
To compute heat fluxes (for visualization, heat exchange coefficient etc) we need to evaluate the gradient of the temperature and populate a vectorvalued (potentially FE) variable with it. Right now the gradient evaluation of the FV variable is not defined for a quadrature point so this returns all zeros at all quadrature points. 

## Design
We need to override the evaluation of the gradient at quadrature points in the variable. 

## Impact
Would make it easy to visualize gradient fields at an input level.

### PR DESCRIPTION
**Pull request by grmnptr**
Closes #23146

### CODE REVIEW

**Review by GiudGiud — DISMISSED** [review]

**Comment by grmnptr** [comment]
Yeah there is some more work to do here apparently.

**Comment by GiudGiud** [comment]
I think you might just need to add some ghosting

With a diffusion kernel for example
Though there might be some debug-like options to add ghosting

File: framework/src/auxkernels/FunctorElementalGradientAux.C [hunk 1]
```diff
@@ -25,6 +25,20 @@ FunctorElementalGradientAuxTempl<is_ad>::validParams()
...
+  // gradient)
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementSideNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC |
+          Moose::RelationshipManagerType::COUPLING,
+      [](const InputParameters & obj_params, InputParameters & rm_params)
+      {
+        rm_params.set<unsigned short>("layers") = obj_params.get<unsigned short>("ghost_layers");
+        rm_params.set<bool>("attach_geometric_early") = false;
```
Comment (lindsayad) [inline framework/src/auxkernels/FunctorElementalGradientAux.C:39]:
```suggestion
```

Comment (lindsayad) [reply framework/src/auxkernels/FunctorElementalGradientAux.C:39]: That shouldn't be necessary

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/auxkernels/FunctorElementalGradientAux.C (hunk 1 above)
Comment (grmnptr) [reply framework/src/auxkernels/FunctorElementalGradientAux.C:39]: Got it, thanks Alex!

File: framework/src/auxkernels/FunctorElementalGradientAux.C [hunk 2]
```diff
@@ -25,6 +25,19 @@ FunctorElementalGradientAuxTempl<is_ad>::validParams()
...
   params.addParam<MaterialPropertyName>(
       "factor_matprop", 1, "A (regular) material property factor to apply on the functor");
+
+  // We need some ghosting for the Finite Volume Fields (we use neighbor information to compute
+  // gradient)
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementSideNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC |
+          Moose::RelationshipManagerType::COUPLING,
```
Comment (lindsayad) [inline framework/src/auxkernels/FunctorElementalGradientAux.C:35]: Oh you should remove coupling from this list as this is not a residual/Jacobian object

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/auxkernels/FunctorElementalGradientAux.C (hunk 2 above)
Comment (grmnptr) [reply framework/src/auxkernels/FunctorElementalGradientAux.C:35]: Thank Alex again! I'll definitely start looking at the relationship manager functions after this, lot to learn!

File: framework/src/auxkernels/FunctorElementalGradientAux.C [hunk 3]
```diff
@@ -25,6 +25,18 @@ FunctorElementalGradientAuxTempl<is_ad>::validParams()
   params.addParam<MooseFunctorName>("factor", 1, "A factor to apply on the functor");
   params.addParam<MaterialPropertyName>(
       "factor_matprop", 1, "A (regular) material property factor to apply on the functor");
+
+  // We need some ghosting for the Finite Volume Fields (we use neighbor information to compute
+  // gradient)
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementSideNeighborLayersAux",
```
Comment (lindsayad) [inline framework/src/auxkernels/FunctorElementalGradientAux.C:33]:
```suggestion
      "ElementSideNeighborLayers",
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/auxkernels/FunctorElementalGradientAux.C (hunk 3 above)
Comment (grmnptr) [reply framework/src/auxkernels/FunctorElementalGradientAux.C:33]: I probably hit every possible mistake on defining this RM haha, will correct this thanks Alex!

Comment (lindsayad) [reply framework/src/auxkernels/FunctorElementalGradientAux.C:33]: I'm really confused why we can compile with this line the way it is

Comment (grmnptr) [reply framework/src/auxkernels/FunctorElementalGradientAux.C]: Okay, I'll look into the RMs to make sure I can add them properly.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**